ablySpec/RealtimeClientChannel.swift

```diff
@@ -181,6 +181,25 @@ class RealtimeClientChannel: QuickSpec {
                     expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Failed), timeout: testTimeout)
                 }
 
+                // RTL4f
+                pending("should transition the channel state to FAILED if ATTACHED ProtocolMessage is not received") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let client = ARTRealtime(options: options)
+                    client.setTransportClass(TestProxyTransport.self)
+                    client.connect()
+                    defer { client.close() }
+
+                    expect(client.connection().state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
+                    let transport = client.transport as! TestProxyTransport
+                    transport.actionsIgnored += [.Attached]
+
+                    let channel = client.channel("test")
+                    channel.attach()
```

Comment:
To make this test more deterministic, I would prefer that you instanced the library with a lower [`realtimeRequestTimeout`](http://docs.ably.io/client-lib-development-guide/features/#DF1b) such as 3s, and then following `attach`, you should expect the state to change to failed in exactly 3s.